src/Contrib/Jaeger/CustomizedTHttpClient.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Jaeger;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+use Thrift\Exception\TTransportException;
+use Thrift\Factory\TStringFuncFactory;
+use Thrift\Transport\THttpClient;
+
+class CustomizedTHttpClient extends THttpClient
+{
+    private ClientInterface $psr18Client;
+
+    private RequestFactoryInterface $requestFactory;
+
+    private StreamFactoryInterface $streamFactory;
+
+    private string $endpointUrl;
+
+    public function setPsr18HttpClient(ClientInterface $client): self
+    {
+        $this->psr18Client = $client;
+
+        return $this;
+    }
+
+    public function setPsr7RequestFactory(RequestFactoryInterface $requestFactory): self
+    {
+        $this->requestFactory = $requestFactory;
+
+        return $this;
+    }
+
+    public function setPsr7StreamFactory(StreamFactoryInterface $streamFactory): self
+    {
+        $this->streamFactory = $streamFactory;
+
+        return $this;
+    }
+
+    public function setEndpointURL(string $endpointUrl): self
+    {
+        $this->endpointUrl = $endpointUrl;
+
+        return $this;
+    }
+
+    /**
+     * Opens and sends the actual request over the HTTP connection
+     *
+     * @throws TTransportException if a writing error occurs
```

Comment:
I don't see any indication that an exception is thrown (and a psr-18 client would not throw a TTransportException)

Reply:
Yup, copy-pasta leftover lol. Removed